integrations/integrations.go

```diff
@@ -0,0 +1,232 @@
+		return fmt.Errorf("failed to start Monomer node in-process: %w", err)
+	}
+
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
+	<-sigCh
+
+	return nil
+}
+
+var _ io.WriteCloser = (*fakeTraceWriter)(nil)
+
+type fakeTraceWriter struct{}
+
+func (f *fakeTraceWriter) Write([]byte) (int, error) {
+	return 0, nil
+}
+
+func (f *fakeTraceWriter) Close() error {
+	return nil
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L624
+func startApp(
+	env *environment.Env,
+	svrCtx *server.Context,
+	appCreator servertypes.AppCreator,
+	opts server.StartCmdOptions,
+) (app servertypes.Application, err error) {
+	home := svrCtx.Config.RootDir
+	db, err := opts.DBOpener(home, server.GetAppDBBackend(svrCtx.Viper))
+	if err != nil {
+		return app, err
+	}
+
+	// TODO: Check if is testnet and implement `testnetify` function
+
+	app = appCreator(svrCtx.Logger, db, &fakeTraceWriter{}, svrCtx.Viper)
+	env.Defer(func() {
+		if localErr := app.Close(); localErr != nil {
+			svrCtx.Logger.Error(localErr.Error())
+		}
+	})
+
+	return app, nil
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L307
+// We modify this function to start a Monomer node in-process instead of a Comet node.
+func startInProcess(
+	env *environment.Env,
+	svrCtx *server.Context,
+	_ serverconfig.Config, //nolint:gocritic // hugeParam
+	clientCtx client.Context, //nolint:gocritic // hugeParam
+	app servertypes.Application,
+	opts server.StartCmdOptions,
+) error {
+	g, ctx := getCtx(svrCtx, true)
+
+	svrCtx.Logger.Info("starting Monomer node in-process")
+	// --- Start the Monomer node ---
+	wrappedApp := &WrappedApplication{app}
+	_, err := startMonomerNode(wrappedApp, env, svrCtx)
+	if err != nil {
+		return err
+	}
+
+	if opts.PostSetup != nil {
+		if err := opts.PostSetup(svrCtx, clientCtx, ctx, g); err != nil {
+			return err
+		}
+	}
+
+	return g.Wait()
+}
+
+// Starts the Monomer node in-process in place of the Comet node. The
+// `MonomerGenesisPath` flag must be set in viper before invoking this.
+func startMonomerNode(wrappedApp *WrappedApplication, env *environment.Env, svrCtx *server.Context) (*node.Node, error) {
+	// --- Bootstrap listeners and DBs ---
+	chainID := monomer.ChainID(1)
+	engineWS, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, err
+	}
+
+	cmtListenAddr := svrCtx.Config.RPC.ListenAddress
+	cmtListenAddr = strings.TrimPrefix(cmtListenAddr, "tcp://")
+	cometListener, err := net.Listen("tcp", cmtListenAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	appdb := dbm.NewMemDB()
+	blockdb := dbm.NewMemDB()
+	txdb := cometdb.NewMemDB()
+	mempooldb := dbm.NewMemDB()
+
+	// --- Load Monomer genesis doc ---
+	monomerGenesisPath := viper.GetString("monomer-genesis-path")
+	svrCtx.Logger.Info("loading Monomer genesis from", "path", monomerGenesisPath)
```

Comment:
nit: log messages should be capitalized
```suggestion
	svrCtx.Logger.Info("Loading Monomer genesis", "path", monomerGenesisPath)
```